lib/common/mainloop.c

```diff
@@ -479,18 +484,19 @@ gio_poll_dispatch_update(enum qb_loop_priority p, int32_t fd, int32_t evts,
         return -ENOMEM;
     }
 
-    /* Because unlike the poll() API, glib doesn't tell us about HUPs by default */
-    evts |= (G_IO_HUP | G_IO_NVAL | G_IO_ERR);
-
-    if (adaptor->is_used) {
+    if (adaptor->source) {
         g_source_remove(adaptor->source);
+        adaptor->source = 0;
```

Comment:
Do we need to set to 0 before calling remove?
ie.

        int tmp = adaptor->source;
        adaptor->source = 0;
        g_source_remove(tmp);